test/mocks/mocks.go

```diff
@@ -407,6 +409,18 @@ func ServiceBindingRequestMock(
 				ResourceRef:      backingServiceResourceRef,
 			},
 		},
+		Status: v1alpha1.ServiceBindingRequestStatus{
+			Conditions: []conditionsv1.Condition{
```

Comment:
The scenario described below is not valid; even if it is a slice of Condition, it should be conceptually treated as a map with keys being the Type. In other words, the conditions collection should not have two elements with the same Type.